### PR TITLE
Add useragent for update calls

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #15381

### Summary of Changes

Add useragent for update calls

##### test 1
- install a outdated extension
- apply this patch
- run the update
- confirm everything works

##### test 2

- install a outdated joomla version
- patch the file manually
- update to the last joomla version

### Expected result

Update works with the useragent checks

### Actual result

Update works but have no useragent

### Documentation Changes Required

None

### CODE REVIEW

File: libraries/joomla/updater/updateadapter.php [hunk 1]
```diff
@@ -239,10 +241,15 @@ protected function getUpdateSiteResponse($options = array())
 
 		$startTime = microtime(true);
 
+		$version    = new JVersion;
+		$httpOption = new Registry;
+		$httpOption->set('userAgent', $version->getUserAgent('Joomla', true, false));
+
+		$http = JHttpFactory::getHttp($httpOption);
```
Comment (mbabker) [inline libraries/joomla/updater/updateadapter.php]: Getting the `JHttp` instance should still be in the try/catch.  `JHttpFactory::getHttp()` will throw an exception if none of our `JHttpTransport` objects can be used (meaning a valid `JHttp` object can't be built).

Comment (zero-24) [inline libraries/joomla/updater/updateadapter.php]: done https://github.com/joomla/joomla-cms/pull/15385/commits/a1a44143786301e1a01829e84a23c4b437db6f87

**Comment by Dade88** [comment]
As from original #15381 I've tested by the following tests:

1. Re-enable mod_security directives on my updates sub-domain
2. Apply changes to a Joomla installation, while another stay 'as is'
3. Install a self-made extension pointing to my update server

Result for the installation with changes receives the ok from the server (Code 200) with a defined User Agent (Mozilla/5.0 Joomla!/3.6.5 Joomla) . From Apache logs:
> 213.251.182.102 dev.tilellit.pro - [19/Apr/2017:01:14:02 +0200] "GET /joomla/adminer.xml HTTP/1.1" 200 902 "-" "Mozilla/5.0 Joomla!/3.6.5 Joomla"

The Joomla install without modification receives an error 403 and no user agent at all. Again Apache log:
> 213.251.182.102 dev.tilellit.pro - [19/Apr/2017:01:13:44 +0200] "GET /joomla/adminer.xml HTTP/1.1" 403 220 "-" "-"

Good Work!

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on f9f3dd5a662e3c0b363984a5fa06414cf3af9ba7<br /><br />Before the patch there is no user agent in the request:
```
"GET /rep/extension.xml HTTP/1.1" 200 720 "-" "-"
```

After the request the user agent is present:
```
"GET /rep/extension.xml HTTP/1.1" 200 720 "-" "Mozilla/5.0 Joomla!/3.7.4-dev Joomla"
```

Works great :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15385">issues.joomla.org/tracker/joomla-cms/15385</a>.</sub>

**Comment by Quy** [comment]
@franz-wohlkoenig RTC

**Comment by ghost** [comment]
RTC as stated above.

**Pull request merged**